tests/test_parse.py

```diff
@@ -234,3 +236,34 @@ class TConfig:
 
         with raises(NotImplementedError):
             parse(TConfig, ["--a", "1", "1", "2"])
+
+
+class TestIgnore:
+    def test_ignore_default(self):
```

Comment:
Should we verify it breaks when an ignored field has no default?

Reply:
Wait no I thought about this before, but this was for those cheeky properties we set at a later time. For example if you are using setters and getters and want to have something prefixed with `_` 
e.g.
```python
@dataclass
class Config:
    _property: int = field(metadata=dict(ignore_arg=True))

@setter
def property(value):
    self._property = value

@getter
def property(value):
    return self._property
```

then we don't want this to fail when nothing is provided when the config is instantiated, right?

Reply:
In the example above, using `parse` would always fail as `_hidden_property` isn't set when calling `__init__`.